### ISSUE TITLE
Enable YamlFrontMatterExtension will break the heading Hr

### ISSUE THREAD

**Issue by yishengjin1413** [lunet-io/markdig#160]
The heading Hr will not work after enable YamlFrontMatterExtension . There is an example here:

---
| Limit identifier | Limit | Comments |
| --- | --- | --- |
| Maximum |200 |beyond 200 c |

**Comment by xoofx** [comment]
Hm, I'm failing to reproduce your issue.

For example the following markdown:

```
---
property: Yaml front matter
---
# This is a heading

---

This is a paragraph
```` 
will correctly remove the frontmatter and output a hr.

The YamlFrontMatter extension expect the document to start with a leading `---`. It is not intended to be used if you don't expect a YamlFrontMatter... or am I missing something?

**Comment by yishengjin1413** [comment]
@xoofx , the issue is when the whole markdown is :

```
---
This is a paragraph
```

Expect it's rendered to HR + paragraph, actually it's  empty, since it rendered as YamlFronMatter.

**Comment by xoofx** [comment]
Fixed by commit fbdb8cf
